src/PSR7/SpecFinder.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenAPIValidation\PSR7;
+
+use cebe\openapi\spec\OpenApi;
+use cebe\openapi\spec\Operation;
+use cebe\openapi\spec\PathItem;
+use cebe\openapi\spec\Response as ResponseSpec;
+use GuzzleHttp\Psr7\ServerRequest;
+use OpenAPIValidation\PSR7\Exception\NoOperation;
+use OpenAPIValidation\PSR7\Exception\NoPath;
+use OpenAPIValidation\PSR7\Exception\NoResponseCode;
+use Psr\Http\Message\ServerRequestInterface;
+
+final class SpecFinder
```

Comment:
This class is now standalone (not built in). Do you think we can do it to traverse parsed OpenAPI schema which was cached?

I have this use case in my head:
1. We built a validator from the yaml file.
2. The schema is cached somewhere and used inside.
3. After validation is complete, can we somehow continue use this finder to maybe find extra details about the matched request or response? 
3.1 RequestValidator returns `OperationAddress`, maybe we can use it to find extra details in the schema?

What do you think?

Reply:
In other words, can we access the parsed `\cebe\openapi\spec\OpenApi` from the validator after the validation is done?

Reply:
Sounds interesting. Can you keep this into another issue, so I can think of it later. I'm sure there is more to improve so it would be several iterations

Reply:
Sure thing! Thank you!